scripts/autotools/libmongoc/CheckCompression.m4

```diff
@@ -4,20 +4,30 @@ PHP_ARG_WITH([mongodb-snappy],
                              [MongoDB: Enable Snappy for compression [default=auto]])],
              [auto],
              [no])
+PHP_MONGODB_VALIDATE_ARG([PHP_MONGODB_SNAPPY], [auto yes no])
 
 PHP_ARG_WITH([mongodb-zlib],
              [whether to enable zlib for compression],
-             [AS_HELP_STRING([--with-mongodb-zlib=@<:@auto/yes/system/bundled/no@:>@],
+             [AS_HELP_STRING([--with-mongodb-zlib=@<:@auto/system/bundled/no@:>@],
                              [MongoDB: Enable zlib for compression [default=auto]])],
              [auto],
              [no])
 
+dnl PHP_ARG_WITH without a value assigns "yes". Treat it like "auto" since we
+dnl fall back to bundled zlib if the system library isn't found
+if test "$PHP_MONGODB_ZLIB" = "yes"; then
+  PHP_MONGODB_ZLIB="auto"
+fi
+
+PHP_MONGODB_VALIDATE_ARG([PHP_MONGODB_ZLIB], [auto system bundled no])
```

Comment:
I made a habit of converting "yes" values _before_ validation so that we never advertise to users that it's a valid value (see the error message produced by `PHP_MONGODB_VALIDATE_ARG`).